commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java

```diff
@@ -0,0 +1,249 @@
+        } finally {
+            if (originalValue != null) {
+                System.setProperty(FrontendUtils.PARAM_FRONTEND_DIR, originalValue);
+            } else {
+                System.clearProperty(FrontendUtils.PARAM_FRONTEND_DIR);
+            }
+        }
+    }
+
+    @Test
+    void init_noFrontendFolder_dontSetInputDirectory() {
+        provider.init(appModel, Map.of());
+
+        assertThat(provider.getInputDirectory()).isNull();
+    }
+
+    @Test
+    void shouldRun_pathIsFile_dontRun() throws IOException {
+        Path file = projectDir.resolve("file.txt");
+        Files.writeString(file, "text");
+
+        assertThat(provider.shouldRun(file, buildConfig())).isFalse();
+    }
+
+    @Test
+    void shouldRun_frontendDirectoryNotExists_dontRun() {
+        Path path = projectDir.resolve("frontend");
+
+        assertThat(provider.shouldRun(path, buildConfig())).isFalse();
+    }
+
+    @Test
+    void shouldRun_defaultPrefix_noCustomClient_dontRun() throws IOException {
+        Path path = projectDir.resolve("frontend");
+        Files.createDirectories(path);
+        configProperties.put("vaadin.endpoint.prefix", "/connect");
+
+        assertThat(provider.shouldRun(path, buildConfig())).isFalse();
+    }
+
+    @Test
+    void shouldRun_defaultPrefix_customClientWithDefaultPrefix_dontRun() throws IOException {
+        Path path = projectDir.resolve("frontend");
+        Files.createDirectories(path);
+        TypescriptClientCodeGenProvider.writeConnectClient("connect", path.resolve("connect-client.ts"));
+        configProperties.put("vaadin.endpoint.prefix", "/connect");
+
+        assertThat(provider.shouldRun(path, buildConfig())).isFalse();
+    }
+
+    @Test
+    void shouldRun_customPrefix_customClientWithSamePrefix_dontRun() throws IOException {
+        Path path = projectDir.resolve("frontend");
+        Files.createDirectories(path);
+        TypescriptClientCodeGenProvider.writeConnectClient("my-prefix", path.resolve("connect-client.ts"));
+        configProperties.put("vaadin.endpoint.prefix", "/my-prefix");
+
+        assertThat(provider.shouldRun(path, buildConfig())).isFalse();
+    }
+
+    @Test
+    void shouldRun_customPrefix_noCustomClient_run() throws IOException {
+        Path path = projectDir.resolve("frontend");
+        Files.createDirectories(path);
+        configProperties.put("vaadin.endpoint.prefix", "/my-prefix");
+
+        assertThat(provider.shouldRun(path, buildConfig())).isTrue();
+    }
+
+    @Test
+    void shouldRun_customPrefix_customClientWithDifferentPrefix_run() throws IOException {
+        Path path = projectDir.resolve("frontend");
+        Files.createDirectories(path);
+        TypescriptClientCodeGenProvider.writeConnectClient("my-prefix", path.resolve("connect-client.ts"));
+        configProperties.put("vaadin.endpoint.prefix", "/different-prefix");
+
+        assertThat(provider.shouldRun(path, buildConfig())).isTrue();
+    }
```

Comment:
Also refactor into one method with signatore of connectClientPrefix and newPrefix (or similiar). And determine the codeGen based on the difference of both? just keep the assertions in each method ?

Reply:
The code has been refactored, maybe now it covers your suggestion.
Otherwise, can you make an example?